client/modules/IDE/components/FileNode.jsx

```diff
@@ -61,6 +61,7 @@ export class FileNode extends React.Component {
     ) {
       this.props.updateFileName(this.props.id, this.originalFileName);
     }
+    this.props.saveFile(this.props.id);
```

Comment:
One issue is that if the file name is invalid (if one of the above if statements is true), then the file name will erroneously get updated.

Reply:
also, i think _only_ the file name should be saved, and not the entire file.